blog/2023-06-20-Deployment-vs-Release.md

```diff
@@ -0,0 +1,56 @@
+
+OK so that's deployments, what about releases? A release is where functionality e.g., a new feature or an upgraded version of an API, is made available to its users. This is all around _when_ the new functionality is presented to the users. The timing and execution of a release is the most important consideration, commonly balancing the needs of the business e.g., getting new feature out that generates revenue, and the expectations of the users e.g., uptime.  It serves to evolve the product, meet user expectations, measure user response for future development refinement, and mitigate risk by controlling the timing and manner of delivery.
+
+The common expectation is a successful deployment is followed immediately by a release, and while this can happen, you might want to deploy a new version without releasing it to your users. There are many ways this can be done, remembering this can be dependant on what you are deploying and the underlying infrastructure of your environment(s):
+
+**Rolling Deployment**: Where the new version is gradually deployed across all instances. As the new version is being deployed, the old versions continue to operate, reducing the chances of system downtime.
+
+**Blue-Green Deployment**: Here two environments (Blue and Green) are maintained. The Blue environment has the live production application, while the Green environment is used for the new version. Once the new version is ready and tested in the Green environment, the routing is switched, and the Green environment becomes live production.
+
+**Canary Deployment**: With canary a new version is initially released to a small subset of servers or users. If everything works well, the new version is progressively rolled out to the rest of the servers or users.
+
+**A/B Testing**: While similar to a Canary deployment, in A/B testing the performance of both versions is compared in order to decide whether the new version is beneficial or not.
+
+**Feature Flags (or Feature Toggles)**: Feature flags allow developers to control the visibility and functionality of a specific feature in a live environment. This means they can turn features on or off, enable them for specific user groups, or roll them out progressively. This deployment strategy enables teams to test new features in the production environment without affecting the entire user base. If an issue arises with the new feature, it can be toggled off immediately, reducing the impact on users. Feature flags provide a safety net and offer flexibility for both deployments and releases.
+
+Deployment and release practices often hit plenty common challenges that can have an impact on their success, especially those practices that have grown organically over time rather than through explicit direction and planning. Recognising these obstacles is the first step towards mitigating them and, towards enhancing the efficiency of these processes. 
```

Comment:
```suggestion
Deployment and release practices often hit plenty of common challenges that can impact their success, especially those practices that have grown organically over time rather than through explicit direction and planning. Recognising these obstacles is the first step towards mitigating them and enhancing the efficiency of these processes. 
```